Prox/Prox/Widgets/ProviderStarImage.swift

```diff
@@ -0,0 +1,82 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+
+// All items in the protocol could be static but I don't know how to do that in swift.
+protocol ProviderStarImageAccessor {
+
+    /// The size, in pixels, of the review score asset. Used for `widthConstraint(for:withViewHeight)`
+    var assetSize: CGSize { get }
+    func image(forScore score: Float) -> UIImage?
+}
+
+extension ProviderStarImageAccessor {
+
+    /// Returns a width constraint that, when activated, constrains the image view to the asset size,
+    /// based on the given height.
+    ///
+    /// HACK: when we scale inside a UIImageView, the UIImageView keeps the size the of the original image
+    /// (rather than taking the scaled image size) so, in this case, the view is too wide and has whitespace
+    /// on either side of the image. For dev speed, we hardcode the asset sizes and set width constraints
+    /// with this method. A proper alternative would be to scale the image outside the image view
+    /// and set the image view with the scaled image.
+    func widthConstraint(for view: UIImageView, withViewHeight viewHeight: CGFloat) -> NSLayoutConstraint {
```

Comment:
This is pretty ugly. Consider importing [this](https://github.com/mozilla-mobile/firefox-ios/blob/716536a06c624dc9424e25edaca012dd50c49bbc/Shared/Extensions/UIImageExtensions.swift#L53) so we can keep this neatly contained in the image class.